test/utils/runners.go

```diff
@@ -1354,7 +1369,24 @@ func NewCustomCreatePodStrategy(podTemplate *v1.Pod) TestPodCreateStrategy {
 // volumeFactory creates an unique PersistentVolume for given integer.
 type volumeFactory func(uniqueID int) *v1.PersistentVolume
 
-func NewCreatePodWithPersistentVolumeStrategy(claimTemplate *v1.PersistentVolumeClaim, factory volumeFactory, podTemplate *v1.Pod) TestPodCreateStrategy {
+// PodTemplate is responsible for creating a v1.Pod.
+type PodTemplate interface {
+	// GetPodTemplate returns a pod template for one out of many different pods.
+	GetPodTemplate(index, count int) (*v1.Pod, error)
+}
+
+// StaticPodTemplate returns an implementation of PodTemplate for a fixed pod that is the same regardless of the index.
+func StaticPodTemplate(pod *v1.Pod) PodTemplate {
+	return (*staticPodTemplate)(pod)
+}
+
+type staticPodTemplate v1.Pod
+
+func (s *staticPodTemplate) GetPodTemplate(index, count int) (*v1.Pod, error) {
```

Comment:
What're the parameters for?

Reply:
* transforming a YAML file with text/template (new) -> dynamicPodTemplate?
* returning a fixed pod instance (what was done before) -> staticPodTemplate?

Reply:
> What're the parameters for?

Added doc comment.

> dynamicPodTemplate/staticPodTemplate

I think `podTemplateFromFile` is more descriptive than `dynamicPodTemplate`. But perhaps I misunderstand the suggestion.

BTW, should code in `test/utils` that is only used by SIG scheduling get moved under a directory owned by that SIG? Not something that is urgent, but as it stands now this isn't general-purpose test helper code. In some cases, like `NewCustomCreatePodStrategy`, scheduler_perf is the only user.

Reply:
Make sense to me to under scheduler_perf folder, make it less spread. We also have utils in test/integration/framework/perf_utils.go, maybe we can group them together, both under scheduler_perf if only used for it.